decidim-accountability/app/models/decidim/accountability/result.rb

```diff
@@ -76,12 +76,24 @@ def comments_have_votes?
         true
       end
 
+      # Public: Whether the object can have new comments or not.
+      def can_participate?(user)
+        can_participate_in_space?(user)
```

Comment:
What do you think about moving this to the concerns? This way logic can be overwritten by each participatory space. Take into account that this should work for all existing participatory spaces.

```ruby
# participable.rb
def can_participate?(user)
  user.present?
end

# models/assembly.rb
def can_participate?(user)
  return true unless participatory_space.private_space?
  return false unless user

  participatory_space.users.include?(user)
end

# has_component.rb
def can_participate_in_space?(user)
  participatory_space.can_participate?(user)
end
```

Reply:
Or even, move this logic to the permissions system, since this is what we're checking here: whether the user has permissions to comment on that space or not. Then you can skip these methods, move this to the permissions system and check the permissions in the API.

Reply:
@mrcasals is there any example of what you are saying? 

>  move this to the permissions system and check the permissions in the API.

Reply:
also, this two methods already existis in concerns. 

```ruby
# participable.rb
def can_participate?(user)
  user.present?
end

# models/assembly.rb
def can_participate?(user)
  return true unless participatory_space.private_space?
  return false unless user

  participatory_space.users.include?(user)
end
```

So, I will extend the functionality using concerns.